AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/databean/AppAction.java

```diff
@@ -0,0 +1,127 @@
+package com.example.appactionvisualizer.databean;
+
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * basic app action class
+ * https://developers.google.com/assistant/app/action-schema#overview
+ * implements Serializable to ensure pass between activities
+ */
+public class AppAction implements Serializable {
+  private static int newId = 0;
+  private static final String TAG = "APPACTION";
+
+  private int actionId;
+  private String appName;
+  private String appPackage;
+  private ArrayList<Action> actions = new ArrayList<>();
+  //todo: add entityset
+
+  public static ArrayList<AppAction> allAppActions= new ArrayList<>();
+  public static Map<ActionType, ArrayList<AppAction>> appActionList = new HashMap<>();
+
+  private AppAction(){}
+
+  public static void parseData() {
+    //TODO: parse from file logic
```

Comment:
I wonder for all the TODOs in the code, can we create corresponding issues to track them
https://docs.github.com/en/github/managing-your-work-on-github

Reply:
fixed